### PR TITLE
[Test] Add ceph-volume starting and stopping OSD

### PR DESCRIPTION
**Pull request by adityaramteke**
Create ceph-volume based OSD.
Start the OSD using ceph-osd command.
Perform OSD in and out operation.
Validate OSDs are getting up and running.

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/aramteke/automation/cephci-run-NGUN98/

### CODE REVIEW

File: cli/ceph/osd/osd.py [hunk 1]
```diff
@@ -67,3 +67,21 @@ def tree(self, epoch=None, states=None, **kw):
...
+    def out(self, id):
+        """Set osd out
+
+        Args:
+            id (str): osd id
+        """
+        cmd = f"{self.base_cmd} out {id}"
+        return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def _in(self, id):
```
Comment (pranavprakash20) [inline cli/ceph/osd/osd.py:80]: Can we remove the _ here? Or any specific reason why its added Aditya?

File: cli/cephadm/cephadm.py [hunk 2]
```diff
@@ -105,3 +105,17 @@ def logs(self, fsid, name):
...
+
+    def ceph_osd(self, id, conf, keyring):
+        """Run ceph-osd command
+
+        Args:
+            id (str): OSD id
+            conf (str): read configuration from the given confirguration file
+            keyring (str): keyring file name
+        """
+        cmd = f"{self.base_cmd} --id {id} --conf {conf} --keyring {keyring}"
```
Comment (pranavprakash20) [inline cli/cephadm/cephadm.py:117]: The command to be executed is ceph-osd right? But i dont see that in the cmd here
The self.base_cmd is cephadm. 
Please let me know if i missed anything

**Review by pranavprakash20 — CHANGES_REQUESTED** [review]

File: cli/ceph/osd/osd.py (hunk 1 above)
Comment (adityaramteke) [reply cli/ceph/osd/osd.py:80]: It wasn't allowing to use python keywords for method; hence I used _.

File: cli/cephadm/cephadm.py (hunk 2 above)
Comment (adityaramteke) [reply cli/cephadm/cephadm.py]: @pranavprakash20 you are right, I miss that adding, thanks for catching it. 
I have corrected this now, and updated the full logs.

File: cli/cephadm/cephadm.py [hunk 3]
```diff
@@ -105,3 +105,17 @@ def logs(self, fsid, name):
         if isinstance(out, tuple):
             return out[0].strip()
         return out
+
+    def ceph_osd(self, id, conf, keyring):
```
Comment (pranavprakash20) [inline cli/cephadm/cephadm.py:109]: one doubt @adityaramteke here, so are these the only params to ceph-osd command?
I mean there should be more isn't it? In that case, can we have these as **args? or similar logic?
Just a thought, what do you think?

File: cli/ceph/osd/osd.py (hunk 1 above)
Comment (pranavprakash20) [reply cli/ceph/osd/osd.py:80]: Got it. @vamahaja need a help here, any other workarounds?

File: cli/cephadm/cephadm.py (hunk 3 above)
Comment (adityaramteke) [reply cli/cephadm/cephadm.py:109]: @pranavprakash20 Yes, that should work, ID is mandatory parameter and rest are additional paramters, I have made the changes accordingly.

File: cli/ceph/osd/osd.py (hunk 1 above)
Comment (adityaramteke) [reply cli/ceph/osd/osd.py:80]: @pranavprakash20, @vamahaja only suggested me to have it this method with _.

Comment (vamahaja) [reply cli/ceph/osd/osd.py:80]: @pranavprakash20 as `in` is inbuild python keyword, use `_in`

**Review by vamahaja — COMMENTED** [review]
LGTM

**Review by pranavprakash20 — APPROVED** [review]
Lgtm

**Review by MohitBis — APPROVED** [review]
LGTM

**Pull request merged**